doc/rst/language/spec.rst

```diff
@@ -9760,22 +9759,33 @@ available for this purpose
 Nilable Class Types
 ~~~~~~~~~~~~~~~~~~~
 
-Variables of class type cannot store ``nil`` unless the class type is
-nilable. To declare a nilable class type, use the ``?`` operator to
-create a nilable type. For example, if ``C`` is a class type, then
+Variables of a class type cannot store ``nil`` and do not have a default
+value unless the class type is nilable. To create a nilable class type,
+use the postfix ``?`` operator. For example, if ``C`` is a class type, then
 ``C?`` indicates the nilable class type with generic management. The
 ``?`` operator can be combined with memory management specifiers as
 well. For example, ``borrowed C?`` indicates a nilable class using the
 ``borrowed`` memory management strategy. Note that the ``?`` operator
 applies only to types.
 
-The postfix ``!`` operator applies to a type or a value. When applied to
-a ``borrowed`` or ``unmanaged`` type, it returns the non-nilable version
-of that type. When applied to an ``owned`` or ``shared`` type, it
-returns the non-nilable borrowed type. When applied to a value, it
-asserts that the value is not ``nil`` and returns that value as a
-non-nilable type. If the value was in fact ``nil``, it halts. It returns
-the borrowed type for ``owned`` or ``shared``.
+A nilable type can also be created with a cast to ``class?``. For example,
+if ``T`` is a class type, then ``T: class?`` indicates its nilable counterpart,
+or ``T`` itself if it is already nilable. ``T: borrowed class?`` modifies ``T``
+as needed to ensure the result is nilable and has the ``borrowed``
+memory management strategy.
+
+To create a non-nilalble class type, apply a cast to ``class`` or
+to a more specific type. For example, if ``T`` is a class type, then
+``T: class`` indicates its non-nilable counterpart, or ``T`` itself
+if it is already non-nilable. ``T: borrowed class`` modifies ``T``
```

Comment:
I wouldn't use "modifies" in this context because one might think it changes what T means. 

Perhaps "``T: borrowed class`` produces the non-nilable ``borrowed`` variant of the class type ``T``."